src/system/newlib_stubs.c

```diff
@@ -56,3 +60,40 @@ int getentropy(void *_buffer, size_t _length) {
 void __sync_synchronize(void) {
 	__sync_synchronize();
 }
+
+int clock_gettime(clockid_t clock_id, struct timespec* tp) {
+	struct timeval tv;
+	int retval = -1;
+
+	switch (clock_id)
+	{
+	case CLOCK_REALTIME:
+		retval = gettimeofday(&tv, NULL);
+		if (!retval) TIMEVAL_TO_TIMESPEC(&tv, tp);
+		break;
+	
+	default:
+		errno = EINVAL;
+		break;
+	}
+
+	return retval;
+}
+
+
+// The value for this variable is added by the common.mk makefile so that the
+// timestamp is up to date with the hot package's latest compilation
+extern char const* _PROS_COMPILE_TIMESTAMP_INT;
+
+int _gettimeofday(struct timeval* tp, void* tzvp) {
+	if (competition_is_connected()) {
+		tp->tv_sec = vexSystemTimeGet() * 1000;
```

Comment:
```suggestion
		tp->tv_sec = vexSystemTimeGet() / 1000;
```

Reply:
The reason we are doing the compile timestamp is because the robot doesn't keep track of the current time unless the robot is connected to field control. We decided that if it's not connected, we could use a compiled timestamp to approximate the current time.

Reply:
Right, I gathered that. But my question is more so what scenarios are you trying to get to enable? I can think of things that this breaks in non-obvious ways, but I can't think of a scenario that this will enable. Keep in mind that these syscalls are used in a lot of scenarios. For example, you're totally breaking any sort of stl and stdlib based timers.